packages/web-app-files/src/components/SideBar/Shares/Collaborators/CollaboratorsEditOptions.vue

```diff
@@ -282,10 +335,153 @@ export default {
     publishChange() {
       this.$emit('optionChange', {
         role: this.selectedRole,
-        permissions: this.additionalPermissions,
+        permissions: this.customPermissions,
         expirationDate: this.enteredExpirationDate
       })
+    },
+
+    selectRole(role) {
+      if (role.name === 'advancedRole') {
+        this.$refs.customPermissionsDrop.show()
+
+        return
+      }
+
+      this.getRecipientsCustomPermissions()
+
+      this.selectedRole = role
+    },
+
+    confirmCustomPermissions() {
+      this.$refs.customPermissionsDrop.hide()
+
+      this.selectedRole = this.advancedRole
+    },
+
+    cancelCustomPermissions() {
+      this.$refs.customPermissionsDrop.hide()
+      this.$refs.rolesDrop.show()
+    },
+
+    getRecipientsCustomPermissions() {
+      this.customPermissions = []
+
+      if (this.collaboratorsPermissions) {
+        for (const permission in this.collaboratorsPermissions) {
+          this.customPermissions.push(permission)
+        }
+      }
+    },
+
+    cycleRoles(event) {
+      // events only need to be captured if the roleSelect element is visible
+      if (!get(this, '$refs.rolesDrop.tippy.state.isShown', false)) {
```

Comment:
little nit pick: better for refactoring to do it like this:
```suggestion
      if (!get(this.$refs.rolesDrop, 'tippy.state.isShown', false)) {
```